### ISSUE TITLE
W0236: Method was expected to be 'non-async', found it instead as 'async' (invalid-overridden-method)

### ISSUE THREAD

**Issue by RomanValov** [pylint-dev/pylint#10464]
### Bug description

It's a duplicate of #3797. Nevertheless I have no option to re-open the bug. Still I would bring the discussion back to the table.

Please check the example below. The `iterate` method is a sync method that expected to return AsyncIterator. Python syntax requires using "async" keyword for the "inlined" async generator implementation (which otherwise remains a sync method). For the comparison please check `aiterate` method which is async method returning async generator.

The code below passes mypy and pyright. It interpreted by python without issues. Nonetheless `pylint` shows a warning:

```python
import asyncio
from abc import ABC, abstractmethod
from collections.abc import AsyncIterator
from collections.abc import Awaitable
from typing import override


class IterBase(ABC):
    @abstractmethod
    def iterate(self) -> AsyncIterator:
        ...
    @abstractmethod
    async def aiterate(self) -> AsyncIterator:
        ...


class Iter(IterBase):
    @override
    async def iterate(self) -> AsyncIterator:
        yield None
    @override
    async def aiterate(self) -> AsyncIterator:
        await asyncio.sleep(0.0)
        return self.iterate()


async def main():
    it = Iter()
    async for item in it.iterate():
        print(item)
    async for item in await it.aiterate():
        print(item)


if __name__ == '__main__':
    asyncio.run(main())
```

### Configuration

```ini

```

### Command used

```shell
pylint its.py
```

### Pylint output

```python
************* Module its
its.py:24:4: W0236: Method 'iterate' was expected to be 'non-async', found it instead as 'async' (invalid-overridden-method)
```

### Expected behavior

No complaints for iterate method

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.12.3 (main, Jun 18 2025, 17:59:45) [GCC 13.3.0]
```

### OS / Environment

Ubuntu 24.04

### Additional dependencies

```python

```

**Comment by mbyrnepr2** [comment]
Similar to these open issues:
#9745 
#5761

**Comment by RomanValov** [comment]
@mbyrnepr2 thanks for the reference. Indeed #5761 is a direct duplicate of my bug and it's open. I've missed it.
So I close my bug and copy my sample to another one.
